src/app/api/services/devicesService.ts

```diff
@@ -261,6 +267,22 @@ export const invokeDirectMethod = async (parameters: InvokeMethodParameters): Pr
     }
 };
 
+export const cloudToDeviceMessage = async (parameters: CloudToDeviceMessageParameters) => {
+    try {
+        if (!parameters.deviceId) {
```

Comment:
Should this throw since it is missing key information instead of returning?

Reply:
I was following the patterns like a no brainer. Was considering removing them together if it makes sense, because if deviceId is not provided, the solution won't be able to compile at all.

Reply:
So as for this PR, I removed this unncessary check. Going to seperate the removal of others in a seperate pr just in case.